samples/SoundBoard/src/main/java/com/google/oboe/samples/soundboard/MusicTileView.java

```diff
@@ -20,16 +36,41 @@
     private boolean[] mIsPressedPerRectangle;
     private Paint mPaint;
     private SparseArray<PointF> mLocationsOfFingers;
-    private NoteListener mNoteListener;
+    private TileListener mNoteListener;
+
+    public interface TileListener {
+        public void onTileOn(int index);
+        public void onTileOff(int index);
+    }
+
+    public class NoteListener implements TileListener {
+        private native void noteOn(long engineHandle, int noteIndex);
+        private native void noteOff(long engineHandle, int noteIndex);
+
+        long mEngineHandle;
+
+        public NoteListener(long engineHandle) {
+            mEngineHandle = engineHandle;
+        }
 
-    public MusicTileView(Context context, ArrayList<Rect> rectangles, NoteListener noteListener) {
+        public void onTileOn(int index) {
+            noteOn(mEngineHandle, index);
+        }
+
+        public void onTileOff(int index) {
+            noteOff(mEngineHandle, index);
+        }
+    }
+
+    public MusicTileView(Context context, ArrayList<Rect> rectangles, long mEngineHandle) {
```

Comment:
MusicTileView should not know anything about the synthesizer.

Reply:
Thanks!